appeals/web/src/server/appeals/appeal-details/application-decision-date/application-decision-date.controller.js

```diff
@@ -0,0 +1,201 @@
+import logger from '#lib/logger.js';
+import { addNotificationBannerToSession } from '#lib/session-utilities.js';
+import { HTTPError } from 'got';
+import { getAppellantCaseFromAppealId } from '../appellant-case/appellant-case.service.js';
+import {
+	changeApplicationHasDecisionDatePage,
+	changeApplicationSetDecisionDatePage
+} from './application-decision-date.mapper.js';
+import { changeApplicationDecisionDate } from './application-decision-date.service.js';
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getChangeApplicationHasDecisionDate = async (request, response) => {
+	return renderChangeApplicationHasDecisionDate(request, response);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderChangeApplicationHasDecisionDate = async (request, response) => {
+	try {
+		const { errors, currentAppeal, apiClient } = request;
+		const appellantCaseData = await getAppellantCaseFromAppealId(
+			apiClient,
+			currentAppeal.appealId,
+			currentAppeal.appellantCaseId
+		);
+
+		const mappedPageContents = changeApplicationHasDecisionDatePage(
+			currentAppeal,
+			appellantCaseData,
+			request.session.applicationDecisionDate
+		);
+
+		delete request.session.applicationDecisionDate;
+
+		return response
+			.status(200)
+			.render('patterns/change-page.pattern.njk', { pageContent: mappedPageContents, errors });
+	} catch (error) {
+		logger.error(error);
+		delete request.session.applicationDecisionDate;
+		if (error instanceof HTTPError && error.response.statusCode === 404) {
+			return response.status(404).render('app/404.njk');
+		} else {
+			return response.status(500).render('app/500.njk');
+		}
+	}
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const postChangeApplicationHasDecisionDate = async (request, response) => {
+	if (request.errors) {
+		return renderChangeApplicationHasDecisionDate(request, response);
+	}
+
+	try {
+		const radio = request.body['application-decision-radio'];
+		const { currentAppeal, apiClient } = request;
+
+		const { appealId, appellantCaseId } = currentAppeal;
+		if (radio === 'yes') {
+			request.session.radio = radio;
```

Comment:
I don't see this being deleted anywhere, is it meant to be after its been used?

Reply:
It was supposed to be in applicationDecisionDate

Reply:
I ended up removing it anyway as it was causing issues with tests.